src/reconcile/rec_write.c

```diff
@@ -1994,16 +2017,19 @@ __rec_split(WT_SESSION_IMPL *session, WT_RECONCILE *r, size_t next_len)
 		/* If the next object fits into this page, we're good to go. */
 		if (r->space_avail >= next_len)
 			return (0);
+		break;
```

Comment:
I agree - that part of the change was wrong. Are you happy with the rest of
it now?
On Feb 6, 2015 4:41 AM, "Keith Bostic" <notifications@github.com> wrote:

> In src/reconcile/rec_write.c
> <https://github.com/wiredtiger/wiredtiger/pull/1631#discussion_r24181840>:
>
> > @@ -1994,16 +2017,19 @@ __rec_split(WT_SESSION_IMPL *session, WT_RECONCILE *r, size_t next_len)
> >  		/* If the next object fits into this page, we're good to go. */
> >  		if (r->space_avail >= next_len)
> >  			return (0);
> > +		break;
>
> @agorrod <https://github.com/agorrod>, this isn't right: if space_avail <
> next_len, we're going to break out of the case and grow the buffer, which
> isn't correct, we may simply be at the end of the page with a handful of
> bytes left and the next key/value pair doesn't fit.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wiredtiger/wiredtiger/pull/1631/files#r24181840>.
>